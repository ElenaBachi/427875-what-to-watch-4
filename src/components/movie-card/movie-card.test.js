import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card.jsx";

const film = {
  title: `Shutter Island`,
  img: `img/shutter-island.jpg`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};
const onFilmImgClick = () => {};
const onMouseEnter = () => {};
const onMouseLeave = () => {};

it(`MovieCard should render correctly`, () => {
  const tree = renderer.create(
      <MovieCard
        film={film}
        onFilmImgClick={onFilmImgClick}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
        isPlaying={false}
      />, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

  expect(tree).toMatchSnapshot();
});
