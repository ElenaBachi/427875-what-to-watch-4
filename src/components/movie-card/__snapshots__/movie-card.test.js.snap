// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
>
  <div
    className="small-movie-card__image"
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Shutter Island
    </a>
  </h3>
</article>
`;
